### ISSUE TITLE
Comprobación del sistema - Chequeo integridad archivos y directorios

### ISSUE THREAD

**Issue by nosolored** [chamilo/chamilo-lms#2191]
En Piwiks (similar a Google Analytics) el área de Administración dispone de una comprobación de sistema para los archivos y directorios.

De esta forma cuando se actualiza y queda alguna carpeta o archivo que pertenece a la versión última se muestra un aviso con instrucciones.

![img01](https://user-images.githubusercontent.com/5868981/31542410-268511e2-b012-11e7-9ef1-cf6f9074ef0c.jpg)
![img02](https://user-images.githubusercontent.com/5868981/31542435-3b353400-b012-11e7-83d8-509ebae6da35.jpg)

**Comment by jmontoyaa** [comment]
We have already something similar:

https://11.chamilo.org/main/admin/system_status.php?

**Comment by nosolored** [comment]
No es exactamente lo mismo. En Piwik se chequea si existen archivos/directorios mezclados de una versión en otra tras realizar una actualización mostrando un aviso y comprobando la integridad de los que se han copiado, si faltan alguno por copiar, si existen directorios que no forman parte de la instalación, plugins que sólo funcionan en versiones anteriores, etc.

A veces cuando actualizan por FTP/SFTP (foros de Chamilo) algunos archivos/directorios de versiones anteriores se mantienen o no se han copiado correctamente o presentan algunas incidencias.

**Comment by ywarnier** [comment]
Sí, eso tiene más que ver con un `php bin/doctrine.php orm:schema-tool:update --dump-sql`.
Para los archivos es otro tema todavía, porque sugiere que deberíamos tener un listado de todos los archivos de Chamilo de algún lado, lo cual podríamos hacer quizás en un script para la generación de los paquetes por publicar.

**Comment by ywarnier** [comment]
Esto siendo dicho, lo ideal sería llegar a un nivel de instalador como Piwik, que es fabuloso: no requiere ninguna intervención en el servidor, ni para descargar los archivos (si es que la configuración está bien hecha desde un inicio y que la actualización no causa problemas).
